File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/SearchTextBox/Features/SearchTextBoxWatermarkTextProperty.cs
using System;
using System.Text;
using System.Data;
using System.Drawing;
using System.ComponentModel;
using System.Collections.Generic;

using Gizmox.WebGUI.Forms;
using Gizmox.WebGUI.Common;

namespace CompanionKit.Controls.SearchTextBox.Features
{
    public partial class SearchTextBoxWatermarkTextProperty : UserControl
    {
        public SearchTextBoxWatermarkTextProperty()
        {
            InitializeComponent();
        }

        /// <summary>
        /// Handles Load event for Page
        /// </summary>
        private void SearchTextBoxWatermarkTextProperty_Load(object sender, EventArgs e)
        {
            // Set watermark
            SetWatermarkText();
        }

        /// <summary>
        /// Handles Click event for Button control
        /// </summary>
        private void button1_Click(object sender, EventArgs e)
        {
            // Set watermark
            SetWatermarkText();
        }

        /// <summary>
        /// Sets watermark text for SearchTextBox control
        /// </summary>
        public void SetWatermarkText()
        {
            searchTextBox1.WatermarkText = textBox1.Text;
        }        
    }
}

File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/SearchTextBox/Features/SearchTextBoxWatermarkTextProperty.vb
Namespace CompanionKit.Controls.SearchTextBox.Features

	Public Class SearchTextBoxWatermarkTextProperty

        Public Sub New()
            ' This call is required by the Visual WebGui UserControl Designer.
            InitializeComponent()
        End Sub

        ''' <summary>
        ''' Handles Load event for Page
        ''' </summary>
        Private Sub SearchTextBoxWatermarkTextProperty_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
            ' Set watermark
            Me.SetWatermarkText()
        End Sub

        ''' <summary>
        ''' Handles Click event for Button control
        ''' </summary>
        Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
            ' Set watermark
            Me.SetWatermarkText()
        End Sub

        ''' <summary>
        ''' Sets watermark text for SearchTextBox control
        ''' </summary>
        Public Sub SetWatermarkText()
            Me.searchTextBox1.WatermarkText = Me.textBox1.Text
        End Sub

	End Class

End Namespace